src/OpenLoco/src/Station.h

```diff
@@ -51,24 +62,34 @@ namespace OpenLoco
         docks,
     };
 
-    namespace StationFlags
+    enum class StationFlags : uint16_t
     {
-        constexpr uint16_t transportModeRail = (1 << 0);
-        constexpr uint16_t transportModeRoad = (1 << 1);
-        constexpr uint16_t transportModeAir = (1 << 2);
-        constexpr uint16_t transportModeWater = (1 << 3);
-        constexpr uint16_t flag_4 = (1 << 4);
-        constexpr uint16_t flag_5 = (1 << 5);
-        constexpr uint16_t flag_6 = (1 << 6);
-        constexpr uint16_t flag_7 = (1 << 7);
-        constexpr uint16_t flag_8 = (1 << 8);
-        constexpr uint16_t allModes = StationFlags::transportModeRail | StationFlags::transportModeRoad | StationFlags::transportModeAir | StationFlags::transportModeWater;
-    }
+        none = 0U,
+        transportModeRail = (1U << 0),
+        transportModeRoad = (1U << 1),
+        transportModeAir = (1U << 2),
+        transportModeWater = (1U << 3),
+        flag_4 = (1U << 4),
+        flag_5 = (1U << 5),
+        flag_6 = (1U << 6),
+        flag_7 = (1U << 7),
+        flag_8 = (1U << 8),
+        allModes = StationFlags::transportModeRail | StationFlags::transportModeRoad | StationFlags::transportModeAir | StationFlags::transportModeWater,
+    };
+    OPENLOCO_ENABLE_ENUM_OPERATORS(StationFlags);
 
-    string_id getTransportIconsFromStationFlags(const uint16_t flags);
+    string_id getTransportIconsFromStationFlags(const StationFlags flags);
 
     struct CargoSearchState;
 
+    enum class CatchmentFlags : uint8_t
+    {
+        none = 0U,
+        flag0 = 0U,
+        flag1 = 1U,
+    };
+    OPENLOCO_ENABLE_ENUM_OPERATORS(CatchmentFlags);
+
```

Comment:
https://github.com/OpenLoco/OpenLoco/blob/59cee31ad175c6f4dbfd03f5e672675d97919ecd/src/OpenLoco/src/Station.cpp#L424

There was a line in `Station.cpp` that said the catchment flags shouldn't be shifted. Is this the correct way to implement these?

Reply:
This specific enum doesn't hold flags, you can remove OPENLOCO_ENABLE_ENUM_OPERATORS but keep it enum class.